src/components/feedback/Toast/useToast.ts

```diff
@@ -1,64 +1,53 @@
 import { useCallback } from 'react'
 import {
   toast,
-  Slide,
-  Bounce,
-  Flip,
-  Zoom,
   ToastContent,
-  ToastOptions as ToastifyToastOptions,
+  ToastOptions,
   TypeOptions as ToastifyTypeOptions
 } from 'react-toastify'
 import { classes } from './ToastStyles'
 import cx from 'classnames'
-import { cond, equals, always, T } from 'ramda'
-
-type ToastOptions = Omit<ToastifyToastOptions, 'transition'> & {
-  transitionType?: 'Slide' | 'Bounce' | 'Zoom' | 'Flip'
-}
+import { getTransitionType, ToastContainerProps } from './types'
+import { renderEnrichedMessage } from './utils'
 
 const useToast = () => {
   return useCallback(
     /**
      *
-     * @param {ToastContent} message The content to be displayed
+     * @param {ToastContent} message The content to be displayed. It can be a string or a React component.
      * @param {('success'|'info'|'warning'|'error')} variant The type of the toast
      * @param {ToastOptions} options Additional options passed to the toast
      */
-    (message: ToastContent, variant?: ToastifyTypeOptions, { transitionType, ...restOptions } = {} as ToastOptions) => {
+    (message: ToastContent, variant?: ToastifyTypeOptions, { transitionType, actions, textProps, ...restOptions } = {} as ToastContainerProps) => {
```

Comment:
options are of type ToastOptions not ToastContainerProps.
revert the wrong casting done here, recreate the ToastOptions type and extend it with the new properties you need.